### PR TITLE
Add localcourse and partnercourse search

### PR DESCRIPTION
**Pull request by alyssaongyx**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/localcourse/LocalCourseSearchCommand.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        this.codePredicate = null;
+        this.namePredicate = namePredicate;
+    }
+
+    @Override
+    public CommandResult execute(SeplendidModel model) throws CommandException {
+        requireNonNull(model);
+
+        model.searchLocalCourses(attribute, codePredicate, namePredicate);
+
```
Comment (dlathyun) [inline src/main/java/seedu/address/logic/commands/localcourse/LocalCourseSearchCommand.java:61]: I think you need to call seplendidmodel.updateSearchList or getSearchList method here to update the list.

File: src/main/java/seedu/address/logic/commands/partnercourse/PartnerCourseSearchCommand.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        this.codePredicate = null;
+        this.namePredicate = namePredicate;
+    }
+
+    @Override
+    public CommandResult execute(SeplendidModel model) throws CommandException {
+        requireNonNull(model);
+
+        model.searchPartnerCourses(attribute, codePredicate, namePredicate);
+
```
Comment (dlathyun) [inline src/main/java/seedu/address/logic/commands/partnercourse/PartnerCourseSearchCommand.java:61]: Same here.

File: src/main/java/seedu/address/logic/parser/localcourse/LocalCourseSearchCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                            LocalCourseSearchCommand.LOCALCOURSE_SEARCH_MESSAGE_USAGE)
+            );
+        }
+
+        SeplendidArgumentMap parameterToArgMap =
+                SeplendidArgumentTokenizer.tokenize(args,
+                        PARAMETER_LOCALCODE, PARAMETER_LOCALNAME);
+
```
Comment (dlathyun) [inline src/main/java/seedu/address/logic/parser/localcourse/LocalCourseSearchCommandParser.java:40]: What are the arguments of search command?

**Review by dlathyun — COMMENTED** [review]
LGTM except few parts that are not too clear for me!

File: src/main/java/seedu/address/logic/commands/partnercourse/PartnerCourseSearchCommand.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    public static final String MESSAGE_SUCCESS = "partner courses searched: %1$s";
+
+    public static final String PARTNERCOURSE_SEARCH_MESSAGE_USAGE = COMMAND_WORD
+            + " : Search partner courses by attributes - partnercode and partnername";
+
+    private final PartnerCourseAttribute attribute;
+
+    private final PartnerCodeContainsKeywordsPredicate codePredicate;
+    private final PartnerNameContainsKeywordsPredicate namePredicate;
```
[the remark this answers is not in the document]
Comment (alyssaongyx) [reply src/main/java/seedu/address/logic/commands/partnercourse/PartnerCourseSearchCommand.java]: I dont think so?

**Review by dlathyun — APPROVED** [review]
LGTM

**Pull request merged**